builtins/amp-img.js

```diff
@@ -239,7 +242,29 @@ export class AmpImg extends BaseElement {
     if (this.getLayoutWidth() <= 0) {
       return Promise.resolve();
     }
-    return this.loadPromise(img);
+    return this.loadPromise(img).then(() => this.warnAgainstOversizedImages_());
```

Comment:
Does this mean for every load of an `amp-img` we are going to measure the Element?

Can this functionality be keyed off development mode?

Reply:
I'm not sure we can make a judgment on whether the image is oversized without loading it and comparing it with the current size. Development mode was precisely the discussion that we wanted to have in the comment thread below.